server/app/views/applicant/ModalFragment.html

```diff
@@ -1,5 +1,5 @@
 <div
-  th:fragment="modalContainer(showProgramIndexLogin, showReviewLogin)"
+  th:fragment="modalContainer(showProgramIndexLogin, showReviewLogin, showUpsellLogin)"
```

Comment:
Why is this a mega-div? Could it be separated into 3 fragments so that callers just have to call:

~{applicant/ModalFragment:: programIndexLogin)

instead of:

~{applicant/ModalFragment:: modalContainer(<!--/* showProgramIndexLogin= */-->true,
                                                             <!--/* showReviewLogin= */-->false,
                                                             <!--/* showUpsellLogin= */-->false)}

?

Reply:
it is mostly because of the dev tools modal which is shown on all pages. I'm open to updating it though. Thus far everything has been contained in the #modal-container element to match the previous implementation, but that's not really necessary

Reply:
ok now that I tried the refactoring, I remember why I originally did it this way. modal.ts (the non north star version) asserts that there is a modal container element containing all modals, and we can't get rid of that until after north star is launched.

So I did the refactoring and I like it better, but I think it has to wait until after north star launch when we can remove modal.ts. Filed an issue here https://github.com/civiform/civiform/issues/7880 with a draft PR linked